### PR TITLE
Cast random.categorical seed to int32 in TF backend

### PR DESCRIPTION
**Pull request by jeffcarp**
src/samplers/top_k_sampler_test.py to run into a compilation error:

```
(OpKernel was found, but attributes didn't match) Requested Attributes: T=DT_FLOAT, Tseed=DT_INT64, output_dtype=DT_INT64){{node while/stateless_categorical/StatelessMultinomial}}
```

### CODE REVIEW

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for working on the fix!

File: keras/src/backend/tensorflow/random.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def uniform(shape, minval=0.0, maxval=1.0, dtype=None, seed=None):
 
 def categorical(logits, num_samples, dtype="int64", seed=None):
     seed = draw_seed(seed)
+    seed = tf.cast(seed, dtype="int32") # Only int32 seeds are supported.
```
[the remark this answers is not in the document]
Comment (jeffcarp) [reply keras/src/backend/tensorflow/random.py]: Good point, thanks - updated.

**Review by hertschuh — APPROVED** [review]

File: keras/src/backend/tensorflow/random_test.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import tensorflow as tf
+
+from keras.src.backend.tensorflow import random
+from keras.src.testing import TestCase
+
+
+class TFRandomTest(TestCase):
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/random_test.py:7]: You'll have to mark this one as tensorflow only, it's failing with the torch backend.

Comment (jeffcarp) [reply keras/src/backend/tensorflow/random_test.py:13]: Done, thanks.

**Review by fchollet — APPROVED** [review]
LGTM, thanks!
